todos-app/src/App/TodoList/TodoListItem/TodoListItem.tsx

```diff
@@ -1,23 +1,45 @@
-import React, { useState, createRef, useEffect } from "react";
-import type { ITodo } from "../../../DataStructure";
+import React, { useState, createRef } from "react";
+import { ITodo } from "../../../DataStructure";
 import "../../../App.css";
 
-interface Props {
+type TodoListItemsProps = {
   todo: ITodo;
-}
+  onToggle(id: string): void;
+};
 
 interface State {
   onEdit: boolean;
 }
 
-const TodoListItem: React.FC<Props> = ({ todo }) => {
-  // const editInput = createRef<HTMLInputElement>();
-  // const init: State = { onEdit: false };
-  // const [state, setState] = useState(init);
+const TodoListItem: React.FC<TodoListItemsProps> = ({ todo, onToggle }) => {
+  const init: State = { onEdit: false };
+  const [state, setState] = useState(init);
+
+  const SwitchStyle = (t: ITodo, onEdit: boolean): string => {
+    switch (true) {
```

Comment:
since switch is not really being used here we can replace with ifs and returns

Reply:
Replaced _switch-case_ statement with _if-else_ statement